File: app/src/main/java/com/example/blizzard/data/database/WeatherDao.java
package com.example.blizzard.data.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.blizzard.data.entities.WeatherDataEntity;

import java.util.List;

@Dao
public interface WeatherDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveWeather(WeatherDataEntity weatherDataEntity);

    @Query("SELECT * FROM weather")
    List<WeatherDataEntity> getAllWeather();


    @Query("SELECT cityName FROM weather WHERE cityName = :cityName")
    LiveData<String> getWeatherForCity(String cityName);

    @Update
    void updateWeatherData(WeatherDataEntity weatherDataEntity);
}

File: app/src/main/java/com/example/blizzard/data/database/WeatherDao.kt
package com.example.blizzard.data.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.blizzard.data.entities.WeatherDataEntity

@Dao
interface WeatherDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveWeather(weatherDataEntity: WeatherDataEntity?)

    @get:Query("SELECT * FROM weather")
    val allWeather: List<WeatherDataEntity?>?

    @Query("SELECT cityName FROM weather WHERE cityName = :cityName")
    fun getWeatherLiveDataForCity(cityName: String?): LiveData<String?>?

    @Query("SELECT * FROM weather WHERE cityName = :cityName")
    fun getWeatherForCity(cityName: String?): WeatherDataEntity?

    @Update
    fun updateWeatherData(weatherDataEntity: WeatherDataEntity?)
}